lib/Reader/abisignature.cpp

```diff
@@ -247,6 +247,20 @@ static Value *getFieldAddress(IRBuilder<> &Builder, Value *Base,
   return Address;
 }
 
+bool ABICallSignature::hasIndirectResultOrArg() const {
```

Comment:
Nit: this should also check for by-value struct arguments. Such arguments will have kind `ABIArgInfo::Direct` and `Arg.getType()->isStructTy()` will return `true`.